server/src/feature/didChangeConfiguration.ts

```diff
@@ -0,0 +1,39 @@
+import { DiagnosticSeverity, DidChangeConfigurationParams, NotificationHandler } from 'vscode-languageserver-protocol';
+
+import { EngineModel } from '../engineModel/EngineModel';
+import Session from '../session';
+
+/**
+ * Map between the string available as option and real `DiagnosticSeverity` enum values.
+ */
+const diagSeverityMap = new Map<string, DiagnosticSeverity>([
+    ['error', DiagnosticSeverity.Error],
+    ['warning', DiagnosticSeverity.Warning],
+    ['information', DiagnosticSeverity.Information],
+    ['hint', DiagnosticSeverity.Hint],
+]);
+
+let engineModel: EngineModel;
+let pathToPredefinedObjectsXml: string;
+let activateParsingProblemsReporting: boolean;
+// By default, the severity is Information.
+export let parsingIssueSeverityLevel: DiagnosticSeverity = DiagnosticSeverity.Information;
+
+export default function(session: Session): NotificationHandler<DidChangeConfigurationParams> {
+    return (change) => {
+        session.onDidChangeConfiguration(change);
+        activateParsingProblemsReporting = change.settings.yseopml.activateParsingProblemsReporting;
+        pathToPredefinedObjectsXml = change.settings.yseopml.pathToPredefinedObjectsXml;
+        // One of the severity levels possible, or `Information`.
+        parsingIssueSeverityLevel =
+            diagSeverityMap.get(change.settings.yseopml.ymlParsingIssueSeverityLevel.toLowerCase()) ||
+            DiagnosticSeverity.Information;
```

Comment:
The default value is kinda written in three places, now, right? Here, in the `export let … = DiagnosticSeverity.Information;`, and in the constructor of `YmlParsingErrorListener`. Seems strange.

Btw, I saw no call of that constructor (`YmlParsingErrorListener`) that was making use of the optional second argument meant to pass the severity level. I don’t quite get how this is handled now.